[11.0] account_multicurrency_revaluation: Fixes and improvements

grindtildeath opened the pull request:
Thanks to @keylor2906 for pointing this out.

Supersedes : #100

grindtildeath committed:
[FIX] Display computation rate on move line label

grindtildeath commented:
@hbto @moylop260 @bodedra @yvaucher @mpanarin  @matthieuchoplin 
Can you guys please pass by for a review to get this merged ?

https://github.com/OCA/account-closing/pull/102 and https://github.com/OCA/account-closing/pull/103 would enjoy your approval as well :wink:

bodedra reviewed (approved):
LGTM (code review)

TDu commented on account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:
```diff
@@ -129,15 +129,15 @@ def _compute_unrealized_currency_gl(self, currency_id, balances, form):
         balance = balances.get('balance', 0.0)
         unrealized_gain_loss = 0.0
         if foreign_balance:
             ctx_rate['revaluation'] = True
```
By moving the assignation of the context up this line would have no effect, no ?

grindtildeath replied:
In such case the context is defined on `currency` var, so that it's not only used on the `compute` call but also when we get the rate here : https://github.com/OCA/account-closing/pull/101/files/3aaae75184fb6c3487270c4f23de8a1008eda73d#diff-2f2adec7633819d0a28eb35fef7fca93R150

TDu replied:
Yes, but does changing the variable used for context after it has been assigned actually impacts the context ?
I don't think so.

grindtildeath replied:
Sorry for the confusion but you are right.
However, as I looked into this context's usage, it was actually not needed, and I added the following commit to remove it.

TDu replied:
Simplified code then :smile:

TDu reviewed (commented):
Also I am not sure what is the difference between applying the context on the browse instead of on the compute ?

grindtildeath committed:
Drop unneeded override of res.currency

The override of res.currency _get_conversion_rate did return 1.0 / from_currency.rate
when 'revaluation' was True in context.
However, as res.currency.rate is defined per company and the context is only
used for a computation to the company currency, the std method from odoo will also
return 1.0 / from_currency.rate, because to_currency is the company currency.
(https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_currency.py#L179)
Therefore, we do not need this override.

grindtildeath committed:
Remove commented line

This line was already commented in v7.0, so we do not need to keep it across all versions

grindtildeath committed:
Make use of api.multi to avoid extra parameter
